File: Assets/Scripts/LuaScripts/src/MainCityUIForm.lua
---
--- 主视图窗口  UI窗体视图层脚本
---

MainCityUIForm = {}
local this = MainCityUIForm


local UI_Manager=CS.SUIFW.UIManager
local uiManager=UI_Manager.GetInstance()

local Lua_Helper=CS.LuaFramework.LuaHelper 
local luaHelper=Lua_Helper.GetInstance()

local transform
local gameobject

--得到实例
function MainCityUIForm.GetInstance()
    return this
 end

function MainCityUIForm.Awake(obj)
    print("------- MainCityUIForm.Awake  -----------");
    gameobject=obj
    transform=obj.transform
end

function MainCityUIForm.Start(obj)
    this.InitView()
end

function MainCityUIForm.InitView()
    --查找UI中按钮
    this.PlayNormalBtn=transform:Find("PlayNormal")--返回transform
    this.PlayNormalBtn=this.PlayNormalBtn:GetComponent("UnityEngine.UI.Button") --返回Button类型
    this.PlayNormalBtn.onClick:AddListener(this.ProcessPlayNormal)

    this.MarketBtn=transform:Find("BtnMarket")--返回transform
    this.MarketBtn=this.MarketBtn:GetComponent("UnityEngine.UI.Button") --返回Button类型
    this.MarketBtn.onClick:AddListener(this.ProcessMarket)

    this.EmallBtn=transform:Find("BtnEmall")--返回transform
    this.EmallBtn=this.EmallBtn:GetComponent("UnityEngine.UI.Button") --返回Button类型
    this.EmallBtn.onClick:AddListener(this.ProcessEmallBtn)
end

function MainCityUIForm.ProcessPlayNormal()
   
    print("执行到 ProcessPlayNormal")  --开始游戏
    --解场景包
    
    luaHelper:DoString("require 'TestStartGame'")
    uiManager:ShowUIForms("GameInfoUIForm")
    uiManager:ShowUIForms("DefenseListUIForm")

    CS.UnityEngine.SceneManagement.SceneManager.LoadScene("Level_One");
end

function MainCityUIForm.ProcessMarket()
   
    print("执行到 ProcessMarket")  
    uiManager:ShowUIForms("MarketUIFrom")
end

function MainCityUIForm.ProcessEmallBtn()
   
    print("执行到 ProcessEmallBtn")  
    uiManager:ShowUIForms("NotificationUIForm")
end

File: Assets/Scripts/DemoProject/MainCityUIForm.cs
/***
 * 
 *    Title: "SUIFW" UI框架项目
 *           主题： 主城窗体
 */
using System.Collections;
using System.Collections.Generic;
using ABFW;
using LuaFramework;
using SUIFW;
using UnityEngine;
using UnityEngine.SceneManagement;

namespace DemoProject
{
    public class MainCityUIForm : BaseUIForm
    {
		public void Awake () 
        {
	        //窗体性质
		    CurrentUIType.UIForms_ShowMode = UIFormShowMode.HideOther;

           

            //事件注册
            //RigisterButtonObjectEvent("BtnMarket",
            //    () => OpenUIForm(ProConst.MARKET_UIFORM)           
            //    );
            //事件注册(弹出公告系统)
            RigisterButtonObjectEvent("BtnEmall",
                () => OpenUIForm(ProConst.NOTIFICATION_UIFORM)
                );
            //开始游戏
            //RigisterButtonObjectEvent("PlayNormal",
            //    () =>
            //    {

            //        LuaHelper.GetInstance().DoString("require 'TestStartGame'");
            //        OpenUIForm(ProConst.GAME_INFO_UIFORM);
            //        OpenUIForm(ProConst.DEFENSE_LIST_UIFORM);

            //        SceneManager.LoadScene("Level_One");
            //        Debug.Log("切换场景完成");


            //    }
            //    );

        }
		
	}
}